tabcmd/commands/datasources_and_workbooks/datasources_and_workbooks_command.py

```diff
@@ -99,25 +99,39 @@ def apply_encoded_filter_value(logger, request_options, value):
     # from apply_options, which expects an un-encoded input,
     # or from apply_url_params via apply_encoded_filter_value which decodes the input
     @staticmethod
-    def apply_filter_value(logger, request_options: TSC.PDFRequestOptions, value: str) -> None:
+    def apply_filter_value(logger, request_options: TSC.RequestOptions, value: str) -> None:
         logger.debug("handling filter param {}".format(value))
         data_filter = value.split("=")
-        request_options.vf(data_filter[0], data_filter[1])
+        # we should export the _DataExportOptions class from tsc
+        request_options.vf(data_filter[0], data_filter[1])  # type: ignore
 
     # this is called from within from_url_params, for each param value
+    # expects either ImageRequestOptions or PDFRequestOptions
     @staticmethod
-    def apply_options_in_url(logger, request_options: TSC.PDFRequestOptions, value: str) -> None:
+    def apply_options_in_url(logger, request_options: TSC.RequestOptions, value: str) -> None:
         logger.debug("handling url option {}".format(value))
         setting = value.split("=")
-        if ":iid" == setting[0]:
+        if len(setting) != 2:
+            logger.warn("Unable to read url parameter '{}', skipping".format(value))
```

Comment:
This warning suggests the option should be skipped. Are we missing an early return?

Reply:
yes, fixed.